src/option.ts

```
@@ -182,10 +198,27 @@ export class Option<T> {
    * Run a closure in a `try`/`catch` and convert it into an Option.
    * If the function throws an Error, an Option containing None will be reutrned.
    *
+   * _Note: Please use `fromAsync` to capture the result of asynchronous closures._
+   * @param {Function} fn The closure to run.
+   * @returns {Option<T>} The result of the closure.
+   */
+  static from<T>(fn: () => T): Option<T> {
+    try {
+      return new Option<T>(fn());
+    } catch (_) {
+      return new Option<T>(none);
+    }
+  }
+
+  /**
+   * Run an asynchronous closure in a `try`/`catch` and convert it into an Option.
+   * If the function throws an Error, an Option containing None will be reutrned.
+   *
+   * _Note: Please use `from` to capture the result of synchronous closures._
    * @param {Function} fn The closure to run.
    * @returns {Promise<Option<T>>} The result of the closure.
    */
-  static async from<T>(fn: (() => T) | (() => Promise<T>)): Promise<Option<T>> {
+  static async fromAsync<T>(fn: () => Promise<T>): Promise<Option<T>> {
     try {
       return new Option<T>(await fn());
     } catch (_) {
```

Do we lose the error by swallowing it in the catch and returning `None`? I wonder if this should be considered "undefined" behavior here and throw?

Thinking of "Producing an invalid value" in ["Behavior considered undefined"](https://doc.rust-lang.org/reference/behavior-considered-undefined.html#behavior-considered-undefined).

---

Ooo interesting, let me have a think!